### PR TITLE
chore: add env var checking to verify-mocha-results

### PR DESCRIPTION
**Pull request by flotwig**
- Closes cypress-io/cypress-internal#426

### User facing changelog

n/a

### Additional details

- Since CircleCI does not scrub env vars from test reports, we do it ourself in verify_mocha_results.
- Renames verify_mocha_results to verify-mocha-results for consistency.
- Also converts verify_mocha_results to async/await and fs.promises instead of fs-extra.
- Also adds tests for verify_mocha_results existing and new behavior.

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### Post-merge tasks

- [ ] Cherry-pick to 10.0 and ensure all test jobs are using verify-mocha-results

### CODE REVIEW

File: scripts/circle-env.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// Returns a map of environment variables defined for this job. `readCircleEnv()` differs from `process.env` - it will
+// only return environment variables explicitly specified for this job by CircleCI project env and contexts
+// NOTE: this Circle API is not stable, and yet it is the only way to access this information.
+async function readCircleEnv () {
+  const taskData = await loadInternalTaskData()
+
+  try {
+    const circleEnv = taskData['Dispatched']['TaskInfo']['Environment']
```
Comment (ryanthemanuel) [inline scripts/circle-env.js:36]: Is this API documented/open source anywhere? Could be useful to have a link to refer to if something gets updated?

Comment (flotwig) [reply scripts/circle-env.js:36]: No, it's an internal API. Do you think a comment would be useful?

```suggestion
    // if this starts failing, try SSHing into a CircleCI job and see what changed in the $CIRCLE_INTERNAL_CONFIG file's schema
    const circleEnv = taskData['Dispatched']['TaskInfo']['Environment']
```

**Review by ryanthemanuel — APPROVED** [review]
Looks good. Left a comment but nothing blocking the PR. Nice work!

File: scripts/unit/verify-mocha-results-spec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+const fs = require('fs').promises
```
Comment (tbiethman) [inline scripts/unit/verify-mocha-results-spec.js:1]: Tests for scripts! 💯 💯 💯

**Review by tbiethman — APPROVED** [review]
This is very cool. And more test coverage is always appreciated!

**Pull request merged**